ChronoStore/test/testreadwrite.cpp

```diff
@@ -0,0 +1,179 @@
+    std::vector<std::pair<uint64_t, uint64_t>> range;
+    DatasetMinMax d = sr.readDatasetRange(STORY, CHRONICLE);
+    if(d.status != 0){
+        std::cout<<"Error retrieving min max range for dataset story: " << STORY<<std::endl; 
+        return -1;
+    }
+
+    for(int64_t i = 0 ; i < NUM_OF_TESTS ; i++){
+
+        auto seed = std::chrono::system_clock::now().time_since_epoch().count();
+        srand(seed);
+
+        int64_t start_val = rand() % (d.MinMax.second - d.MinMax.first + 1) + d.MinMax.first;
+        int64_t end_val = rand() % (d.MinMax.second - start_val + 1) + start_val;
+        range.push_back(std::make_pair(start_val, end_val));
+    }
+
+    int64_t totalDataRead = 0;
+    auto start_time = std::chrono::high_resolution_clock::now();
+    for(int i = 0 ; i < range.size() ; i++){
+        DatasetReader dr = sr.readFromDataset(range[i], STORY, CHRONICLE);
+        if(dr.status != 0 || dr.eventData.size() == 0){
+            // std::cout<<"Dataset read test failed on test: "<<i+1<<"\t"<<range[i].first<<"\t"<<range[i].second<<std::endl;
+            return -1;
+        }
+        totalDataRead += dr.eventData.size()*sizeof(Event);
+    }
+
+    // Calculate the duration in seconds
+    auto end_time = std::chrono::high_resolution_clock::now();
+    auto duration = std::chrono::duration_cast<std::chrono::seconds>(end_time - start_time).count();
+    std::cout << "\nTime taken: " << duration << " seconds\nRead bandwidth: "<<(totalDataRead/(1024*1024*duration)) << " MB/second\n";
+
+
+    int result = remove(CHRONICLE);
+    if(result != 0) {
+        std::cout << CHRONICLE << " not found!" <<std::endl;
+        return -1;
+    }
+    return 0;
+}
+
+void generateData(std::string fileName, int64_t startTimeStamp){
+
+    int64_t num_entries = 100000; // each chunk generates 100000 events
+    std::vector<std::pair<int64_t, std::string>> entries(num_entries);
+    int64_t num = startTimeStamp;
+
+    // Const string in all entries for testing purpose
+    std::string str = "R7YYRjIL82qc1jHXk80GXGvzmaGS4e5U2kzyfhpdbkxbY1fXUcfoBdGpXsE5AdjjViRe1e4519bZhQjn2zvZKC64g6B309k6l6wx";
+    int j = 0;
+    for (int i = 0; i < num_entries; i++) {
+    entries[i] = std::make_pair(num, str);
+    num+=2;
+    }
+
+    // Open file for writing
+    std::ofstream out_file(fileName, std::ios::out | std::ios::app);
+
+    // Append sorted entries to file
+    for (const auto& entry : entries) {
+    out_file << entry.first << " " << entry.second << " ";
+    }
+
+    // Close file
+    out_file.close();
+}
+
+int main(int argc, char* argv[]) {
+// The test program generates five storychunk data containing 1000000 events in every chunk and writes to the storydataset for the chronicle H5. further the read and datarange queries are executed to test.
+    int storyChunksCount = 5;
+    std::string filename = "testChunk";
+    int64_t startTimeStamp = 1000;
+    for(int count = 0 ; count < storyChunksCount ; count++){
```

Comment:
can you make these command line arguments?

Reply:
Added storyChunkCount and startTimeStamp as cmd argument to test executable